docs/topics/testing/tools.txt

```diff
@@ -1473,47 +1473,64 @@ your test suite.
 
         self.assertFieldOutput(EmailField, {'a@a.com': 'a@a.com'}, {'aaa': ['Enter a valid email address.']})
 
-.. method:: SimpleTestCase.assertFormError(response, form, field, errors, msg_prefix='')
+.. method:: SimpleTestCase.assertFormError(form, field, errors, msg_prefix='')
 
-    Asserts that a field on a form raises the provided list of errors when
-    rendered on the form.
+    Asserts that a field on a form raises the provided list of errors.
 
-    ``response`` must be a response instance returned by the
-    :class:`test client <django.test.Response>`.
+    ``form`` is a ``Form`` instance. The form must be
+    :ref:`bound <ref-forms-api-bound-unbound>` but not necessarily
+    validated (``assertFormError()`` will automatically call ``full_clean()``
+    on the form).
+
+    ``field`` is the name of the field on the form to check. To check the form's
+    :meth:`non-field errors <django.forms.Form.non_field_errors>`, use
+    ``field=None``.
+
+    ``errors`` is a list of all the error strings that the field is expected to
+    have. You can also pass a single error string if you only expect one error
+    which means that ``errors='error message'`` is the same as
+    ``errors=['error message']``.
 
-    ``form`` is the name the ``Form`` instance was given in the template
-    context of the response.
+    .. versionchanged:: 4.1
 
-    ``field`` is the name of the field on the form to check. If ``field``
-    has a value of ``None``, non-field errors (errors you can access via
-    :meth:`form.non_field_errors() <django.forms.Form.non_field_errors>`) will
-    be checked.
+        In older versions, using an empty error list with ``assertFormError()``
+        always pass, regardless of whether the field had any errors or not.
```

Comment:
`will always pass`, or `would always pass`, or `always passed` — I would favour `would always pass`, which has a sense of `not current`.